redpanda/values.yaml

```diff
@@ -13,135 +13,201 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# values.yaml
-#
 # This file contains values for variables referenced from yaml files in the templates directory.
 #
 # For further information on Helm templating see the documentation at:
-#   https://helm.sh/docs/chart_template_guide/values_files/
+#  https://helm.sh/docs/chart_template_guide/values_files/
+
+# Common parameters
 #
-# For an explanation of Redpanda configuration values:
-#   https://vectorized.io/docs/advanced-configuration/
+# Override redpanda.name template
+nameOverride: ""
+# Override redpanda.fullname template
+fullnameOverride: ""
+# Default kuberentes cluster domain
+clusterDomain: cluster.local
+# Additional labels added to all Kubernetes objects
+commonLabels: {}
 
+# Redpanda parameters
+#
 image:
   repository: vectorized/redpanda
+  # Redpanda version. This determines the installed version (not Chart.appVersion)
+  tag: v21.11.16
   # The imagePullPolicy will default to Always when the tag is 'latest'
-  tag: v21.11.15
-
-# Additional labels to apply to all Kubernetes resources created by this chart.
-labels: {}
-  # app.kubernetes.io/part-of: my-app
-
-clusterDomain: cluster.local
-
-# See https://vectorized.io/docs/configuration/
-config:
-  # If you have a license key please set it here or you can request one
-  # here: https://vectorized.io/redpanda
-  license_key: ""
-
-  # Update with your organization name
-  organization: ""
-
-  redpanda:
-    enable_sasl: false
-    admin:
-      # The port for the admin server.
-      #
-      # Metrics are served off of this port at /metrics.
-      port: 9644
-      address: 0.0.0.0
-      name: admin
-      external:
-        enabled: false
-
-    # Kafka API listeners
+  pullPolicy: IfNotPresent
+# Your license key (optional)
+license_key: ""
+# Your organization name (optional)
+organization: ""
+#
+# Authentication
+auth:
+  #
+  # SASL configuration
+  sasl:
+    enabled: false
+    # user list
+    users:
+      - admin
+  #
+  # TLS configuration
+  tls:
+    # Enable global TLS, which turns on TLS by default for all listeners
+    # Each listener must include a certificate name in its TLS section
+    # Any certificates in auth.tls.certs will still be loaded if enabled is false
+    # This is because listeners may enable TLS individually (see listeners.<listener name>.tls.enabled)
+    enabled: true
+    # list all certificates below, then reference a certificate's name in each listener (see listeners.<listener name>.tls.cert)
+    certs:
+      cert1:
+        caEnabled: true
+        # duration: 43800h
+        path: /etc/tls/certs/cert1
```

Comment:
This is a big change, decoupling the certificates from each listener. Now listeners and certificates can be added separately and associated with each other, ensuring certificates are only defined once. These certs are in a map so they can be retrieved via name.